src/ripple/nodestore/impl/DatabaseShardImp.cpp

```diff
@@ -428,7 +428,14 @@ DatabaseShardImp::importShard(
     }
 
     it->second.shard = std::move(shard);
-    finalizeShard(it->second, true, lock);
+    it->second.state = ShardInfo::State::postImport;
+
+    [[maybe_unused]] auto
+        canAttempt = it->second.attemptHashConfirmation();
```

Comment:
Why is this called here?

Reply:
Just an extra precaution. Removed with your patch.